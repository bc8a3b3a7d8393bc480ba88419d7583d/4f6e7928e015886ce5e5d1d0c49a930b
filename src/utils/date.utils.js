import moment from 'moment';

const supportedDateFormats = {
  ddddDDMMMYYYY: 'dddd, DD MMM YYYY',
};

const format = (date, dateFormat) => moment(date).format(dateFormat);

export default {
  format,
  supportedDateFormats,
};
