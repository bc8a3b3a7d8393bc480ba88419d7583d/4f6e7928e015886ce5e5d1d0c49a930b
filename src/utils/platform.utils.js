import { Platform } from 'react-native';
import DeviceInfo from 'react-native-device-info';
import { equals } from 'ramda';

const isDEV = () => equals(process.env.NODE_ENV, 'development');
const isIOS = () => equals(Platform.OS, 'ios');
const isAndroid = () => equals(Platform.OS, 'android');
const iPhoneX = 'iPhone X';
const isIPhoneX = () => isIOS() && equals(DeviceInfo.getModel(), iPhoneX);

export { isDEV, isIOS, isAndroid, isIPhoneX, iPhoneX };
