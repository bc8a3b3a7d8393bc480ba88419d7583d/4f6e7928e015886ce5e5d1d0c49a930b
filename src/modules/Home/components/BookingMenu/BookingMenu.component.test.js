import React from 'react';
import { shallow } from 'enzyme';

import { testSnapshots } from '../../../../utils/test.utils';

import Component from './BookingMenu.component';

describe('Modules/Home - BookingMenu', () => {
  const componentProps = {
    onPressFlight: jest.fn(),
  };

  const render = props => shallow(<Component {...props} />);

  testSnapshots(Component, [
    {
      description: 'basic render',
      props: componentProps,
    },
  ]);

  test('get position of component', () => {
    const wrapper = render(componentProps);
    const instance = wrapper.instance();

    instance.onLayout({
      nativeEvent: {
        layout: {
          y: 1,
          height: 1,
        },
      },
    });

    expect(instance.position).toEqual(2);
  });
});
