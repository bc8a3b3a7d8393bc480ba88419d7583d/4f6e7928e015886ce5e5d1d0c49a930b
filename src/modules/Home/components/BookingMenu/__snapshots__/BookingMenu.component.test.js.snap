// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modules/Home - BookingMenu snapshots basic render 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "flexDirection": "row",
      "flexWrap": "wrap",
      "marginTop": -45,
      "padding": 10,
    }
  }
>
  <BookingMenuItem
    image="flight"
    key="home.bookingMenu.flights"
    onPress={[MockFunction]}
    title="home.bookingMenu.flights"
  />
  <BookingMenuItem
    image="hotel"
    key="home.bookingMenu.hotels"
    onPress={[MockFunction]}
    title="home.bookingMenu.hotels"
  />
  <BookingMenuItem
    image="attractionsAndActivities"
    key="home.bookingMenu.attractionsAndActivities"
    onPress={[MockFunction]}
    title="home.bookingMenu.attractionsAndActivities"
  />
  <BookingMenuItem
    image="airportTransport"
    key="home.bookingMenu.airportTransport"
    onPress={[MockFunction]}
    title="home.bookingMenu.airportTransport"
  />
  <BookingMenuItem
    image="flightStatus"
    key="home.bookingMenu.flightStatus"
    onPress={[MockFunction]}
    title="home.bookingMenu.flightStatus"
  />
  <BookingMenuItem
    image="priceAlerts"
    key="home.bookingMenu.priceAlerts"
    onPress={[MockFunction]}
    title="home.bookingMenu.priceAlerts"
  />
  <BookingMenuItem
    image="myPoints"
    key="home.bookingMenu.myPoints"
    onPress={[MockFunction]}
    title="home.bookingMenu.myPoints"
  />
</View>
`;
